chain/indexer/src/block.rs

```diff
@@ -100,12 +100,14 @@ where
         let (db_latest_block, checksum) = self.db.get_last_indexed_block(None).await?;
         info!("Loaded indexer state at block #{db_latest_block} with checksum: {checksum}");
 
-        let latest_block_in_db = self.cfg.start_block_number.max(db_latest_block as u64);
+        let next_block_to_process = if self.cfg.start_block_number < db_latest_block as u64 {
+            // If some prior indexing took place already, avoid reprocessing the `db_latest_block`
+            db_latest_block as u64 + 1
```

Comment:
What if not all the logs in the `db_latest_block` got processed? Actually should the db store both the `latest_block` and the `latest_log_index`?

Reply:
Actually not, because either all logs are processed in a block and only then the last indexed block value is updated in the db, or none at all. All happens in a single db transaction.